src/parkapi_sources/converters/apcoa/validators.py

```diff
@@ -0,0 +1,162 @@
+"""
+Copyright 2024 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+from datetime import datetime, timezone
+from decimal import Decimal
+from enum import Enum
+from typing import Optional
+from zoneinfo import ZoneInfo
+
+from validataclass.dataclasses import validataclass
+from validataclass.exceptions import ValidationError
+from validataclass.validators import (
+    DataclassValidator,
+    EnumValidator,
+    IntegerValidator,
+    ListValidator,
+    Noneable,
+    NumericValidator,
+    StringValidator,
+    UrlValidator,
+)
+
+from parkapi_sources.models.enums import ParkingSiteType
+from parkapi_sources.validators import SpacedDateTimeValidator
+
+
+class ApcoaParkingSpaceType:
+    WOMEN_SPACES = 'Women Spaces'
+    FAMILY_SPACES = 'Family Spaces'
+    CARSHARING_SPACES = 'Carsharing Spaces'
+    DISABLED_SPACES = 'Disabled Spaces'
+    EV_CHARGING_BAYS = 'EV Charging Bays'
+    EV_CHARGING = 'EV Charging'
+    TOTAL_SPACES = 'Total Spaces'
+    ELECTRIC_CAR_CHARGING_SPACES = 'Electric Car Charging Spaces'
+    ELECTRIC_CAR_FAST_CHARGING_SPACES = 'Electric Car Fast Charging Spaces'
+    BUS_OR_COACHES_SPACES = 'Bus/Coaches Spaces'
+    CAR_RENTAL_AND_SHARING = 'Car rental & sharing (weekdays from 8am to 8pm)'
+    PICKUP_AND_DROPOFF = 'PickUp&DropOff (weekdays from 8pm to 8am)'
+
+
+class ApcoaCarparkType(Enum):
+    MLCP = 'MLCP'
+    OFF_STREET_OPEN = 'Off-street open'
+    OFF_STREET_UNDERGROUND = 'Off-street underground'
+    ON_STREET = 'On-street'
+    OPEN_SURFACE = 'Open Surface'
+
+    def to_parking_site_type_input(self) -> ParkingSiteType:
+        # TODO: find out more details about this enumeration for a proper mapping
+        return {
+            self.MLCP: ParkingSiteType.OFF_STREET_PARKING_GROUND,
+            self.OFF_STREET_OPEN: ParkingSiteType.OFF_STREET_PARKING_GROUND,
+            self.OFF_STREET_UNDERGROUND: ParkingSiteType.UNDERGROUND,
+            self.ON_STREET: ParkingSiteType.ON_STREET,
+            self.OPEN_SURFACE: ParkingSiteType.OFF_STREET_PARKING_GROUND,
+        }.get(self, ParkingSiteType.OTHER)
+
+
+class ApcoaNavigationLocationType:
+    DEFAULT = 'default'
+    CAR_ENTRY = 'CarEntry'
+
+
+class ApcoaOpeningHoursWeekdays(Enum):
+    MONDAY = 'Monday'
+    TUESDAY = 'Tuesday'
+    WEDNESDAY = 'Wednesday'
+    THURSDAY = 'Thursday'
+    FRIDAY = 'Friday'
+    SATURDAY = 'Saturday'
+    SUNDAY = 'Sunday'
+
+    def to_osm_opening_day_format(self) -> Enum:
+        return {
+            self.MONDAY: 'Mo',
+            self.TUESDAY: 'Tu',
+            self.WEDNESDAY: 'We',
+            self.THURSDAY: 'Th',
+            self.FRIDAY: 'Fr',
+            self.SATURDAY: 'Sa',
+            self.SUNDAY: 'Su',
+        }.get(self, None)
+
+
+class ApcoaOpeningHoursTime(Enum):
```

Comment:
Why do you have an Enum here? What should be enumerated?

Reply:
it's now used as string directly